src/vtbackend/Functions.h

```diff
@@ -618,10 +623,82 @@ inline auto const& functions() noexcept
     return funcs;
 }
 
+class SupportedSequences
+{
+  public:
+    SupportedSequences()
+    {
+        for (auto [index, seq]: crispy::indexed(functions()))
+        {
+            _supportedSequences[index] = seq;
+        }
+        crispy::sort(
+            _supportedSequences,
+            [](FunctionDefinition const& a, FunctionDefinition const& b) constexpr { return compare(a, b); });
+        _lastIndex = _supportedSequences.size();
+    }
+
+    [[nodiscard]] gsl::span<FunctionDefinition const> getAllSequences() const noexcept
+    {
+        return gsl::span<FunctionDefinition const>{_supportedSequences.cbegin(), _supportedSequences.cend()};
+    }
+
+    [[nodiscard]] gsl::span<FunctionDefinition const> getSupportedSequencesAtCurrentVTLevel() const noexcept
+    {
+        return gsl::span<FunctionDefinition const>{_supportedSequences.cbegin(),
+                                                   _supportedSequences.cbegin() + _lastIndex};
+    }
+
+    void updateVTLevel(VTType vt) noexcept
+    {
+        auto itr = std::partition(
+            _supportedSequences.begin(),
+            _supportedSequences.end(),
+            [vt](const FunctionDefinition& value) noexcept { return value.conformanceLevel <= vt; });
+
+        _lastIndex = std::distance(_supportedSequences.begin(), itr);
+        gsl::span<FunctionDefinition> availableDefinition{_supportedSequences.begin(), _lastIndex};
+        crispy::sort(
+            availableDefinition,
+            [](FunctionDefinition const& a, FunctionDefinition const& b) constexpr { return compare(a, b); });
+    }
+
+    void disableSequence(FunctionDefinition seq) noexcept
+    {
+        auto seqIter = std::find(_supportedSequences.begin(), _supportedSequences.begin() + _lastIndex, seq);
+        if (seqIter != _supportedSequences.begin() + _lastIndex)
+        {
+            std::rotate(seqIter, seqIter + 1, _supportedSequences.end());
+            --_lastIndex;
+        }
+    }
+
+    void enableSequence(FunctionDefinition seq) noexcept
+    {
+        auto* seqIter = std::find(_supportedSequences.begin() + _lastIndex, _supportedSequences.end(), seq);
+        if (seqIter != _supportedSequences.end())
+        {
+            // Maybe could be done better since rest of the data is sorted
+            std::iter_swap(_supportedSequences.begin() + _lastIndex, seqIter);
+            gsl::span<FunctionDefinition> arr(_supportedSequences.begin(),
+                                              _supportedSequences.begin() + _lastIndex + 1);
+            crispy::sort(arr, [](FunctionDefinition const& a, FunctionDefinition const& b) constexpr {
+                return compare(a, b);
+            });
+            ++_lastIndex;
+        }
+    }
+
+  private:
+    std::array<FunctionDefinition, functions().size()> _supportedSequences;
+    size_t _lastIndex;
```

Comment:
```suggestion
    size_t _lastIndex = functions().size();
```
Turns out to be that by default (when looking at the ctor above.

Also, `_lastIndex` does NOT point to the last valid item but to the first item next to the last element. Therefore, we probably want to call that `_endIndex` or similar, to denote this?